src/cmd/linuxkit/run_qemu.go

```diff
@@ -525,13 +526,19 @@ func buildQemuCmdline(config QemuConfig) (QemuConfig, []string) {
 	if config.NetdevConfig == "" {
 		qemuArgs = append(qemuArgs, "-net", "none")
 	} else {
-		mac := generateMAC()
-		qemuArgs = append(qemuArgs, "-net", "nic,model=virtio,macaddr="+mac.String())
+		var macRemoveRegExp = regexp.MustCompile(",((?:[0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2})|((?:[0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2}),")
+		var macRegExp = regexp.MustCompile("([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})")
+		var mac = macRegExp.FindString(config.NetdevConfig)
```

Comment:
this would end in a string split by ',', checking each element of the resulting array if it is a valid mac and then rejoin the array without the mac with a ',' delimeter. i would be fine with this solution.